cajita/src/Cajita_SparseIndexSpace.hpp

```diff
@@ -500,23 +503,39 @@ class SparseIndexSpace
 
   private:
     //! block index space, map tile ijk to tile
-    BlockIndexSpace<MemorySpace, cell_bits_per_tile_dim, cell_num_per_tile_dim,
-                    cell_num_per_tile, hash_type, key_type, value_type>
+    BlockMap<MemorySpace, cell_bits_per_tile_dim, cell_num_per_tile_dim,
+             cell_num_per_tile, hash_type, key_type, value_type>
         _block_id_space;
     //! tile index space, map cell ijk to cell local No inside a tile
-    TileIndexSpace<cell_bits_per_tile_dim, cell_num_per_tile_dim,
-                   cell_num_per_tile>
+    TileMap<cell_bits_per_tile_dim, cell_num_per_tile_dim, cell_num_per_tile>
         _tile_id_space;
     //! space size (global), channel size
     Kokkos::Array<int, rank> _min;
     Kokkos::Array<int, rank> _max;
 };
 
+//---------------------------------------------------------------------------//
+// Creation function for SparseMap from GlobalMesh<SparseMesh>
+template <class Scalar, typename MemorySpace,
```

Comment:
Reorder these such that `MemorySpace` comes before `Scalar`. That way if a user wants to just use the defaults then only `MemorySpace` is needed and `Scalar` will be deduced from the `global_mesh` parameter. In the current order a user would still need to provide `Scalar` even though it could be deduced.